aries_cloudagent/wallet/keys/manager.py

```diff
@@ -0,0 +1,105 @@
+"""Multikey class."""
+
+from ...core.profile import Profile
+from ..base import BaseWallet
+from ..key_type import ED25519
+from ..util import b58_to_bytes, bytes_to_b58
+from ...utils.multiformats import multibase
+from ...wallet.error import WalletNotFoundError
+
+DEFAULT_ALG = "ed25519"
+ALG_MAPPINGS = {
+    "ed25519": {"key_type": ED25519, "prefix_hex": "ed01", "prefix_length": 2}
+}
+
+
+class MultikeyManagerError(Exception):
+    """Generic MultikeyManager Error."""
+
+
+class MultikeyManager:
+    """Class for managing wallet keys."""
+
+    def __init__(self, profile=Profile):
+        """Initialize the MultikeyManager."""
+        self.profile = profile
+
+    def _multikey_to_verkey(self, multikey, alg=DEFAULT_ALG):
+        """Transform multikey to verkey."""
+        prefix_length = ALG_MAPPINGS[alg]["prefix_length"]
+        public_bytes = bytes(bytearray(multibase.decode(multikey))[prefix_length:])
+        return bytes_to_b58(public_bytes)
+
+    def _verkey_to_multikey(self, verkey, alg=DEFAULT_ALG):
+        """Transform verkey to multikey."""
+        prefix_hex = ALG_MAPPINGS[alg]["prefix_hex"]
+        prefixed_key_hex = f"{prefix_hex}{b58_to_bytes(verkey).hex()}"
+        return multibase.encode(bytes.fromhex(prefixed_key_hex), "base58btc")
+
+    async def kid_exists(self, wallet, kid):
```

Comment:
To this point, the convention has been to accept a session rather than a session based component